src/Rule/Ip.php

```diff
@@ -222,165 +204,142 @@ public function __construct()
      * Example:
      *
      * ```php
-     * [
-     *      'ranges' => [
+     * (new Ip())->ranges([
      *          '192.168.10.128'
      *          '!192.168.10.0/24',
      *          'any' // allows any other IP addresses
-     *      ]
-     * ]
+     *      ]);
      * ```
      *
      * In this example, access is allowed for all the IPv4 and IPv6 addresses excluding the `192.168.10.0/24` subnet.
      * IPv4 address `192.168.10.128` is also allowed, because it is listed before the restriction.
-     * @property array the IPv4 or IPv6 ranges that are allowed or forbidden.
-     * See [[setRanges()]] for detailed description.
+     * @return static
      */
-    public function setRanges($ranges)
+    public function ranges(array $ranges)
     {
-        $this->ranges = $this->prepareRanges((array)$ranges);
+        $this->ranges = $this->prepareRanges($ranges);
+        return $this;
     }
 
-    /**
-     * @return array The IPv4 or IPv6 ranges that are allowed or forbidden.
-     */
-    public function getRanges()
+    public function getRanges(): array
     {
         return $this->ranges;
     }
 
-    public function validateValue($value): Result
+    /**
+     * @return static
+     */
+    public function allowIpv4(bool $value = true)
     {
-        $result = new Result();
-
-        if (!$this->ipv4 && !$this->ipv6) {
-            throw new \RuntimeException('Both IPv4 and IPv6 checks can not be disabled at the same time');
-        }
+        $this->allowIpv4 = $value;
+        return $this;
+    }
 
-        $result = $this->validateSubnet($value);
-        if (is_array($result)) {
-            $result[1] = array_merge(['ip' => is_array($value) ? 'array()' : $value], $result[1]);
-            return $result;
-        }
+    /**
+     * @return static
+     */
+    public function allowIpv6(bool $value = true)
+    {
+        $this->allowIpv6 = $value;
+        return $this;
+    }
 
-        return $result;
+    /**
+     * @return static
+     */
+    public function allowNegation(bool $value = true)
```

Comment:
```suggestion
    public function allowNegation(bool $value = true): self
```

Reply:
How about allowNegation and disallowNegation?

Reply:
Same w/ other allow* methods.

Reply:
> How about allowNegation and disallowNegation?

like it.

Reply:
added